server/neptune/workflows/internal/deploy/terraform/runner.go

```diff
@@ -74,6 +74,7 @@ func (r *WorkflowRunner) Run(ctx workflow.Context, deploymentInfo DeploymentInfo
 		Root:         r.buildRequestRoot(deploymentInfo.Root, diffDirection, scope),
 		DeploymentID: id.String(),
 		Revision:     deploymentInfo.Commit.Revision,
+		WorkflowMode: terraformActivities.Deploy,
```

Comment:
I'm pretty sure adding workflow arguments does not cause a non-deterministic error, the same way modifying activity inputs don't, but wanted to highlight this just in case.